web/modules/components/src/lib/modal.ts

```diff
@@ -44,12 +50,24 @@ export class Modal<T extends typeof SvelteComponent> {
    *   DOM when the dialog is open.
    */
   constructor(component: T, open = false, lazy = true) {
-    this.component = component
     this._open = open
 
     const modals = document.getElementById("modals")
     if (!modals) throw new Error("Modals container not found")
 
+    this.loading = this.loadDialog(component, open, lazy)
+  }
+
+  /** Imports and constructs the dialog. */
+  private async loadDialog(component: T, open: boolean, lazy: boolean) {
+    component = component instanceof Promise ? (await component).default : component
```

Comment:
Would this be worth moving into a helper?

Reply:
Dealing with `import()` is always a mess, mostly due to bundlers. I'd rather just handle situations like this manually.